pages/api/auth/signin.js

```diff
@@ -0,0 +1,51 @@
+require("dotenv").config();
+import bcrypt from "bcrypt";
+import sql from "~/lib/postgres";
+import * as yup from "yup";
+import jwt from "jsonwebtoken";
+
+const schema = yup.object().shape({
+  email: yup.string().email().required(),
+  hash_password: yup.string().required(),
+});
+
+const handler = async ({ method, body: { email, password } }, res) => {
+  if (method === "POST") {
+    const user = {
+      email: email.toLowerCase(),
+      hash_password: password,
+    };
+    const schemaValid = await schema.isValid(user);
+    if (schemaValid) {
+      const userExists = await sql`SELECT * FROM users WHERE email=${email}`;
+      if (userExists.count > 0) {
+        try {
+          const userValid = await bcrypt.compare(
+            password,
+            userExists[0].hash_password
+          );
+          if (userValid) {
+            const jsonToken = await jwt.sign(
+              user,
+              process.env.ACCESS_TOKEN_SECRET
+            );
+            return res.send(jsonToken);
+            return res.status(200).json({ message: "Success" });
+          } else {
+            return res
+              .status(400)
+              .json({ message: "Email or password incorrect" });
```

Comment:
Related to my comment below, I _think_ (could be wrong) you'll only hit this if the password is incorrect, because it looks to me like you've checked the e mail is stored on line 21.

Reply:
Thank you for the feedback, is still work in progress for now and needs testing after, I pushed it to get feedback for the register page